esp-hal/README.md

```diff
@@ -7,15 +7,33 @@
 
 `no_std` HAL implementations for the peripherals which are common among Espressif devices. Implements a number of the traits defined by [embedded-hal](https://github.com/rust-embedded/embedded-hal).
 
-This crate should not be used directly; you should use one of the device-specific HAL crates instead:
-
-- [esp32-hal](../esp32-hal/README.md)
-- [esp32c2-hal](../esp32c2-hal/README.md)
-- [esp32c3-hal](../esp32c3-hal/README.md)
-- [esp32c6-hal](../esp32c6-hal/README.md)
-- [esp32h2-hal](../esp32h2-hal/README.md)
-- [esp32s2-hal](../esp32s2-hal/README.md)
-- [esp32s3-hal](../esp32s3-hal/README.md)
+[esp-hal] encompasses the hardware abstraction layer across multiple Espressif devices atop peripheral access crate. To select a concrete chip, correct chip feature has to be used. The current list of chip features is `esp32, esp32c2, esp32c3, esp32c6, esp32h2, esp32p4, esp32s2, esp32s3`.
+
+The package tree is as follows:
+- [devices] holds the information about which `peripherals` are supported by the chip HW, `symbols` created by the maintainers to simplify writing drivers for multiple chips, and `efuse` description.
+- [ld] contains `linker-scripts` defining the layout of the chip's memory sections, specify the location of various symbols (such as ROM functions, for example).
+- [src] is a place, where _all_ peripheral drivers for main (high-power) core are placed. Each peripheral is separated into it's own source file(s).
+
+### Quickstart
+
+After [configuring] the environment and [tooling] you can build and run our [examples]
+- make sure, you are in [examples] directory.
+- use `cargo +$TOOLCHAIN $CHIP_FEATURE --bin=$EXAMPLE_NAME`, for example, to run a `hello_world` example for `ESP32-C3` using `nightly` toolchain: `cargo +nightly esp32c3 --bin=hello_world`
+
+> [!NOTE]
+>
+> Please note, that some peripherals miss on some chip HW. Always double-check the [devices] of the chip you want to use or description of example.
```

Comment:
When using the xtask it will tell you if your chip is not supported, so I don't think we need that last bit if we are going to use the xtask exclusively.